beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -310,6 +307,34 @@ func (vs *Server) proposeGenericBeaconBlock(ctx context.Context, req *ethpb.Gene
 		}
 	}
 
+	// Broadcast the new block to the network.
+	blkPb, err := blk.Proto()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get protobuf block")
+	}
+	if err := vs.P2P.Broadcast(ctx, blkPb); err != nil {
```

Comment:
Shouldn't we check if we can properly receive the block before we rebroadcast?

Reply:
This is the status quo. Reason being validators and beacon node are trusted. Broadcast first to optimize latency. We can make an argument to change it with beacon api